packages/astro/src/integrations/index.ts

```diff
@@ -141,7 +141,11 @@ export async function runHookConfigSetup({
 					updatedSettings.watchFiles.push(path instanceof URL ? fileURLToPath(path) : path);
 				},
 				addDevOverlayPlugin: (entrypoint) => {
-					updatedSettings.devOverlayPlugins.push(entrypoint);
+					// TODO add a deprecation warning in Astro 5.
+					hooks.addDevToolbarApp(entrypoint);
```

Comment:
I'd rather deprecate in Astro 5 and break in Astro 6. Stick with our policy. It might feel better for us to deprecate now, but that almost certainly means users getting warnings 4 days for now. Since this change is our fault, we're the ones that need to eat our vegetables.